### PR TITLE
ftjam: add test

### PR DESCRIPTION
**Pull request by bw-yuhogg**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

Pretty much identical to #4376, except that this is for `ftjam` rather than original `jam`.

### CODE REVIEW

File: Formula/ftjam.rb [hunk 1]
```diff
@@ -18,4 +18,23 @@ def install
...
+      #include <stdio.h>
+
+      int main(void)
+      {
+          printf("FtJam Test\\n");
+          return 0;
+      }
+    EOS
+
+    assert_match /Cc ftjamtest.o/, shell_output(bin/"jam").strip
```
Comment (DomT4) [inline Formula/ftjam.rb]: If this doesn't need to be a regex just use `""` rather than `//`.

Comment (rwhogg) [inline Formula/ftjam.rb]: It doesn't actually need to be a regex; about to force-push.

Want me to submit a PR for `jam` to change that one too once this is merged, or should we just leave it?

Comment (zmwangx) [inline Formula/ftjam.rb]: There are too many of them, so I'd prefer we either fix them en mass or fix only as we update.

Comment (DomT4) [inline Formula/ftjam.rb]: Don't mind either way, but fixing as things are updated seems cool.

File: Formula/ftjam.rb [hunk 2]
```diff
@@ -18,4 +18,23 @@ def install
...
+      #include <stdio.h>
+
+      int main(void)
+      {
+          printf("FtJam Test\\n");
+          return 0;
+      }
+    EOS
+
+    assert_match "Cc ftjamtest.o", shell_output(bin/"jam").strip
```
Comment (zmwangx) [inline Formula/ftjam.rb]: Don't really need to `.strip`, we're justing matching a substring.

File: Formula/ftjam.rb [hunk 3]
```diff
@@ -18,4 +18,23 @@ def install
...
+    assert_equal "FtJam Test", shell_output("./ftjamtest").strip
```
Comment (zmwangx) [inline Formula/ftjam.rb]: What about just matching the unstripped output

```rb
assert_equal "FtJam Test\n", shell_output("./ftjamtest")
```

Sorry for being picky.

Comment (rwhogg) [inline Formula/ftjam.rb]: Eh, no worries.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/057773b7a2ca430769f6dc0143fcdffa046a7a4a. Thank you @rwhogg 🙇

**Pull request merged**